### PR TITLE
Add OpenVINO MLFlow benchmarking tutorial

### PR DESCRIPTION
**Pull request by stefannica**
Preview available here: https://stefannica.github.io/docs/dev/tutorials/openvino-mlflow/

### CODE REVIEW

File: docs/tutorials/openvino-mlflow.md [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+
+You'll deploy a FuseML workflow to train a CNN model on the CIFAR-10 dataset and then serve it using three different ML serving tools and configurations. Afterwards, you'll run some performance testing on the deployed inference servers and interpret the results.
+
+## Setup
+
+### Infrastructure
+
+This tutorial uses a GKE cluster explicitly configured to have Cascade or Skylake generation Intel processors, which are better suited for Intel accelerated ML workloads, but it should work on any other Kubernetes cluster. 
+
+NOTE: the FuseML workflow used in this tutorial includes some steps that make use of explicit CPU resource requests. For this reason, it is recommended that the Kubernetes nodes have at least 2 vCPUs each, preferably 4, otherwise the workflow steps might be stuck because they cannot be scheduled due to resource and placement constraints. The alternative, if an acceptable number of CPUs cannot be ensured for the Kubernetes nodes, is to edit the FuseML workflow to reduce or even eliminate the explicit CPU resource requirements. The cluster used in this tutorial has 2 nodes, each with 4 vCPUs.
```
Comment (flaviodsr) [inline docs/tutorials/openvino-mlflow.md:17]: mkdocs offers text blocks specifically for those kings of notes. I would recommend using that here, you can see how it works in: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

File: docs/tutorials/openvino-mlflow.md [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+
+<figure markdown> 
+  ![GKE Node Pool](img/openvino-mlflow/gke2.png)
+  <figcaption>GKE Cluster Node Pool Configuration Settings</figcaption>
+</figure>
+
+<figure markdown> 
+  ![GKE Nodes](img/openvino-mlflow/gke3.png)
+  <figcaption>GKE Cluster Node Configuration Settings</figcaption>
+</figure>
```
Comment (flaviodsr) [inline docs/tutorials/openvino-mlflow.md:35]: I wonder if there is way to make it so when you click on the image it opens it on the original resolution. As it is now, it is very difficult to see the information on the image as the it is small, unless I increase the zoom in the page. Another option might be to only show the the significant parts of the image instead of the whole screen.

File: docs/tutorials/openvino-mlflow.md [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+    Workflow "mlflow-multi-e2e" assigned to codeset "cifar10/keras"
+    ```
+
+4. Wait for the workflow run to finish running:
+
+    By assigning the workflow to the codeset, a workflow run will be created. You can check the status
+    of the workflow run by running the following command:
+
+    ```bash
+    fuseml workflow list-runs
```
Comment (flaviodsr) [inline docs/tutorials/openvino-mlflow.md:266]: I noticed that for some commands you include a `$` before the call, while others don't. I think it would be good have all command calls on the same way.

File: docs/tutorials/openvino-mlflow.md [hunk 4]
```diff
@@ -0,0 +1,611 @@
...
+      ![Workflow WIP In Tekton](img/openvino-mlflow/tekton-wip.png)
+      <figcaption>View of the Workflow in Progress in the Tekton Dashboard</figcaption>
+    </figure>
+
+    Aside from the long training time, note that since this is the first time the workflow is running, it will build a docker image including the software dependencies for training the model which may take a while. However, consecutive runs will skip that step as long as the dependencies are kept the same.
+
+    <figure markdown> 
+      ![Workflow Done In Tekton](img/openvino-mlflow/tekton-done.png)
+      <figcaption>View of the Completed Workflow in the Tekton Dashboard</figcaption>
+    </figure>
```
Comment (flaviodsr) [inline docs/tutorials/openvino-mlflow.md:290]: Some comment about those images, it is really hard to see anything without zooming in.

File: docs/tutorials/openvino-mlflow.md [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+
+The list of FuseML applications include a URL to query the model for predictions. In addition to the URLs, since our workflow includes a KServe prediction service, you can also check the deployed models through the KServe
+dashboard (`http://kserve-web-app.<FUSEML_DOMAIN>/`, in our case `http://kserve-web-app.34.118.93.249.nip.io/`) which also includes more detailed information, such as the status of the deployment, logs, etc.
+
+For example:
+
+<figure markdown> 
+  ![KSserve Dashboard](img/openvino-mlflow/kserve.png)
+  <figcaption>KFServing Dashboard</figcaption>
+</figure>
```
Comment (flaviodsr) [inline docs/tutorials/openvino-mlflow.md:335]: Some comment about those images, it is really hard to see anything without zooming in.

**Review by flaviodsr — CHANGES_REQUESTED** [review]
Just a couple of suggestions, but overall looks great!

File: docs/tutorials/openvino-mlflow.md [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
++--------------+-----------------------+---------+------+-------------------------------+-------------------------------------------+----------------------------+
+```
+
+## Training & Serving the Model
+
+We will be training a
+[Convolutional Neural Network (CNN)](https://developers.google.com/machine-learning/glossary/#convolutional_neural_network)
+to classify [CIFAR-10 images](https://www.cs.toronto.edu/~kriz/cifar.html) using the Keras Sequential API.
+The complete original code for the model training is available [here](https://github.com/fuseml/examples/tree/main/codesets/mlflow/keras). Our model architecture has been changed slightly, to make it converge faster and to yield better results.
+For an overview of the changes, please refer to [this Machine Learning Mastery article](https://machinelearningmastery.com/how-to-develop-a-cnn-from-scratch-for-cifar-10-photo-classification/).
```
Comment (jsuchome) [inline docs/tutorials/openvino-mlflow.md:209]: It's actually not clear from this paragraph what changes are you mentioning. _Our model architecture has been changed slightly_ : compared to what exactly?

File: docs/tutorials/openvino-mlflow.md [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+
+### Training & Serving using FuseML
+
+The following steps describe how to use FuseML to train the Keras CIFAR-10 model and serve it.
+
+1. Clone the `fuseml/examples` repository and register the `keras` example code as a FuseML codeset:
+
+    ```bash
+    $ git clone --depth 1 https://github.com/fuseml/examples.git
```
Comment (jsuchome) [inline docs/tutorials/openvino-mlflow.md:218]: We usually clone specific branch of examples repo, to keep the tutorial stable.

File: docs/tutorials/openvino-mlflow.md [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+    ```bash
+    $ fuseml workflow create examples/workflows/mlflow-multi-e2e.yaml
+    Workflow "mlflow-multi-e2e" successfully created
+    ```
+
+
+3. Assign the `mlflow-multi-e2e` workflow to the `keras` codeset:
+
+    ```bash
+    $ fuseml workflow assign -c keras -p cifar10 -n mlflow-multi-e2e
```
Comment (jsuchome) [inline docs/tutorials/openvino-mlflow.md:256]: Could you use long names of the options so it is more clear to the user what do they mean?

File: docs/tutorials/openvino-mlflow.md (hunk 1 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:17]: done

File: docs/tutorials/openvino-mlflow.md (hunk 2 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:35]: done

File: docs/tutorials/openvino-mlflow.md (hunk 3 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:266]: done

File: docs/tutorials/openvino-mlflow.md (hunk 4 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:290]: I added a featherlight extension to help with that

File: docs/tutorials/openvino-mlflow.md (hunk 5 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:335]: fixed

File: docs/tutorials/openvino-mlflow.md (hunk 6 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md]: Ok, I rephrased it, hopefully it makes more sense now.

File: docs/tutorials/openvino-mlflow.md (hunk 7 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:217]: the main branch is implicit, I don't need to specify it as an argument. When we backport, we'll have to change it (e.g. to v3).

File: docs/tutorials/openvino-mlflow.md (hunk 8 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md]: done

File: docs/tutorials/openvino-mlflow.md [hunk 9]
```diff
@@ -0,0 +1,611 @@
...
+
+    <figure markdown> 
+      ![Workflow Done In Tekton](img/openvino-mlflow/tekton-done.png)
+      <figcaption>View of the Completed Workflow in the Tekton Dashboard</figcaption>
+    </figure>
+
+
+### Validating the Deployed Model
+
+Before querying the served model for predictions, lets take a look at MLflow for detailed information about
```
Comment (jsuchome) [inline docs/tutorials/openvino-mlflow.md:295]: Typo: `let's`

File: docs/tutorials/openvino-mlflow.md [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+    
+    - the workflow has 3 different predictor steps, each one targeting a different serving solution (OVMS, KServe and Seldon Core). They all take in the same model artifact that comes out of the training step.
+    - the OVMS predictor needs the model to be converted to the Intel IR (Intermediate Representation) format. This is facilitated by a converter step.
+    - there are explicit CPU resource requests specified for the training and OVMS serving steps. They are meant to speed up the training process and to give the OVMS predictor access to an entire vCPU. Feel free to edit the workflow and change the resource requirements to fit your scenario. Note that the Kubernetes cluster needs to have these resources available, otherwise the workflow will fail because the step containers cannot be scheduled.
+    - the default values for the workflow input parameters specify training the model for 30 epochs. This will take approximately 40 minutes and yield an accuracy close to 0.8. You can modify these parameters to experiment with different values.
```
Comment (jsuchome) [inline docs/tutorials/openvino-mlflow.md:243]: Could you also mention that you've added explicit application names and that if you haven't done it, they'd be generated automatically (to something less nice)? 
Either here or at the point where you do the application list..

File: docs/tutorials/openvino-mlflow.md (hunk 9 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md]: fixed

File: docs/tutorials/openvino-mlflow.md (hunk 10 above)
Comment (stefannica) [reply docs/tutorials/openvino-mlflow.md:244]: great idea, I missed that ! Done

**Review by jsuchome — APPROVED** [review]

**Review by flaviodsr — APPROVED** [review]
